chips/e310x/src/chip.rs

```diff
@@ -76,35 +75,32 @@ impl<A: 'static + Alarm<'static>> kernel::Chip for E310x<A> {
     }
 
     fn service_pending_interrupts(&self) {
-        let mut reenable_intr = FieldValue::<u32, mie::Register>::new(0, 0, 0);
-
         loop {
             let mip = CSR.mip.extract();
 
             if mip.is_set(mip::mtimer) {
                 unsafe {
                     timer::MACHINETIMER.handle_interrupt();
                 }
-                reenable_intr += mie::mtimer::SET;
             }
             if mip.is_set(mip::mext) {
                 unsafe {
                     Self::handle_plic_interrupts();
                 }
-                reenable_intr += mie::mext::SET;
             }
 
             if !mip.matches_any(mip::mext::SET + mip::mtimer::SET) {
                 break;
             }
         }
 
-        // re-enable any interrupt classes which we handled
-        CSR.mie.modify(reenable_intr);
+        // Re-enable all MIE interrupts that we care about. Since we looped
+        // until we handled them all, we can re-enable all of them.
+        CSR.mie.modify(mie::mext::SET + mie::mtimer::SET);
     }
 
     fn has_pending_interrupts(&self) -> bool {
-        unsafe { plic::has_pending() }
+        CSR.mip.matches_any(mip::mext::SET + mip::mtimer::SET)
```

Comment:
This change confuses me, will this match list need to grow as more interrupts are supported?

Reply:
Specifically, your comment said
> Not only look at the plic, but also the MIE register for if interrupts have occurred.
But this change seems to look at MIE instead of rather than in addition to?

Reply:
PLIC == `mext`

Reply:
This looks good. Can OT have the same change as well?